src/ProductConstructionService/ProductConstructionService.DependencyFlow/PullRequestBuilder.cs

```diff
@@ -199,10 +200,22 @@ public async Task<string> GenerateCodeFlowPRTitleAsync(
         return await CreateTitleWithRepositories($"[{targetBranch}] Source code changes from ", [update.SubscriptionId]);
     }
 
-    public async Task<string> GenerateCodeFlowPRDescriptionAsync(SubscriptionUpdateWorkItem update)
+    public async Task<string> GenerateCodeFlowPRDescriptionAsync(
+        SubscriptionUpdateWorkItem update,
+        string previousSourceCommit)
     {
-        var build = await _barClient.GetBuildAsync(update.BuildId)
-            ?? throw new Exception($"Failed to find build {update.BuildId} for subscription {update.SubscriptionId}");
+
+        var build = await _barClient.GetBuildAsync(update.BuildId);
+
+        string sourceDiffText;
+        if (previousSourceCommit != null && !string.IsNullOrEmpty(build.GitHubRepository))
+        {
+            sourceDiffText = $"[View Source Diff]({build.GitHubRepository}/compare/{previousSourceCommit}..{build.Commit})";
```

Comment:
Can the anchor text show something more meaningful? E.g. the SHAs?

Something like this?

```suggestion
            sourceDiffText = $"[{previousSourceCommit}..{build.Commit}]({build.GitHubRepository}/compare/{previousSourceCommit}..{build.Commit})";
```

Reply:
I had it like that and the link was extremely long and it looked quite bad. Let me know if this compact link is fine or if we should include the long one with the SHAs anyway.

Reply:
It's also another custom to shorten the SHAs to 7 characters (GitHub and AzDO both do it in their UI). We even have a method for that somewhere. So maybe we could do these short SHAs?